### DISCUSSION TITLE
CLR reference search path

### DISCUSSION THREAD

**Discussion by bcalco** [leppie/IronScheme#105] [Discussions]
When using `(clr-reference ...)` are directories in the `(library-path)` also searched for assemblies, or is it strictly using the .NET rules for finding them? Is there a way to refer to the full path of an assembly one wishes to load, like F#'s `#r` directive does? Documentation suggests only a symbol of the DLL name or a string of the fully qualified assembly name. I would like to load an assembly from a resolved path as well.

In my application, users create projects with a certain folder structure and I'd like to make it easy for them to install DLLs "locally" by some definition of the term that meets with search path heuristics when referring to .NET assemblies. For example, I'd like during a "build" of my project to install the assemblies in a project-local "bin" directory.

**Comment by leppie** [comment]
Oops, I never saw this.

`clr-reference` probably requires the library to be local or in the GAC. 

I do see it either takes a symbol or string. Eg:

```
(clr-reference System.Windows.Forms)
(clr-reference "System.Windows.Forms")
```

This will first try load as:
```
var aname = AssemblyName.GetAssemblyName(assname);
Assembly.Load(aname);
```
If that fails, tt has some handling for custom Assembly resolvers if you provide them. 

Then it will try as a last effort:
```
Assembly.LoadWithPartialName(assname);
```

So whatever those calls allow or does, will work.

**Reply by leppie** [reply]
The Scheme and CLR stuff are kept separate for most part. As mentioned in the exception handling discussion, the requirements are quite different.
